sympy/physics/mechanics/tests/test_rigidbody.py

```diff
@@ -125,6 +125,11 @@ def test_parallel_axis():
     Ip_expected = inertia(N, Ix + m * b**2, Iy + m * a**2,
                           Iz + m * (a**2 + b**2), ixy=-m * a * b)
     assert Ip == Ip_expected
+    A = ReferenceFrame('A')
+    A.orient_axis(N, N.z, 1)
+    assert (R.parallel_axis(p, A).to_matrix(A) -
+            Ip_expected.to_matrix(A)).simplify() == zeros(3, 3)
```

Comment:
Why does the symengine let `simplify` return `None`, while normal python engine returns `zeros(3, 3)`?

Reply:
SymPy's Matrix class is mutable and therefore not a subclass of Basic. Unlike everything else in SymPy the mutable Matrix class has a `simplify` method that does in-place simplification and returns None in the same way that `list.sort` does an in place sort and returns `None`. On the other hand if you `sympify` a Matrix then you get an ImmutableMatrix which is an instance of Basic and has a simplify method that returns a new simplified ImmutableMatrix without modifying the original (immutable) matrix.

Different operations will convert a mutable Matrix into an ImmutableMatrix or not. SymEngine also has the same mutable and immutable matrix concept but doesn't handle the distinction in the same way as SymPy. Most likely you're hitting up against:
https://github.com/symengine/symengine.py/issues/363

Reply:
Thank you for the clarification and it indeed seems to be the same issue, so good to know it is already open.